modules/core/src/test/java/org/apache/ignite/internal/processors/cache/datastructures/GridCacheAbstractDataStructuresFailoverSelfTest.java

```diff
@@ -442,22 +447,40 @@ public void testSemaphoreNonFailoverSafe() throws Exception {
 
             assertEquals(0, sem.availablePermits());
 
-            new Timer().schedule(new TimerTask() {
-                @Override public void run() {
-                    stopGrid(NEW_IGNITE_INSTANCE_NAME);
+            CountDownLatch sync = new CountDownLatch(1);
+
+            FutureTask<Boolean> task = new FutureTask<>(new Callable<Boolean>() {
+                @Override public Boolean call() throws Exception {
+                    try {
+                        // Notify main thread.
+                        sync.countDown();
```

Comment:
This will not compile under jdk7.
Please fix code and check correctness under jdk7